juju/juju
# fix: use access service in control socket worker

The control socket worker previously used state to add the read permission for the metrics user. It now uses the access service.





## Checklist




## QA steps
IN PROGRESS (having some issues with microk8s)
```
juju bootstrap microk8s test-control-socket
juju switch controller
juju deploy prometheus-k8s p --trust
juju relate p controller

juju users # should include `juju-metrics-r0`

```

## Links

**Jira card:** [JUJU-6525](https://warthogs.atlassian.net/browse/JUJU-6525)
